### PR TITLE
Fix of [BeamMobsim.iteration] is not unique

### LINKED ISSUE #2514: Bug: actor name [BeamMobsim.iteration] is not unique!
During CI got an exception that [actor name is not unique](https://beam-ci.tk/job/beam-forced-prs/2137/consoleFull):
```
00:23:26 A PersonAgent
00:23:26 - should allow scheduler to set the first activity (9 milliseconds)
00:23:26 - should demonstrate a complete trip, throwing MATSim events (80 milliseconds)
00:23:26 - should know how to take a walk_transit trip when it's already in its plan (29 milliseconds)
00:23:26 17:23:26.235 [PersonAgentSpec-akka.actor.default-dispatcher-5] ERROR beam.agentsim.agents.PersonAgent - BeamAgent Got Shutdown. This means actorRef.stop() was called externally, e.g. by supervisor because of an exception. In state PerformingActivity
00:23:26 
00:23:26 17:23:26.235 [PersonAgentSpec-akka.actor.default-dispatcher-5] ERROR beam.agentsim.agents.PersonAgent - State: PerformingActivity Event: StopEvent(Shutdown,PerformingActivity,BasePersonData(0,None,List(),Vector(),None,None,,0,false,0.0,0,None))
00:23:26 17:23:26.235 [PersonAgentSpec-akka.actor.default-dispatcher-14] ERROR beam.agentsim.agents.PersonAgent - Events leading up to this point:
00:23:26 	LogEntry(Uninitialized,BasePersonData(0,None,List(),Vector(),None,None,,0,false,0.0,0,None),TriggerWithId(InitializeTrigger(0),1))
00:23:26 	LogEntry(Initialized,BasePersonData(0,None,List(),Vector(),None,None,,0,false,0.0,0,None),TriggerWithId(ActivityStartTrigger(0),2))
00:23:26 	LogEntry(PerformingActivity,BasePersonData(0,None,List(),Vector(),None,None,,0,false,0.0,0,None),TriggerWithId(ActivityEndTrigger(10),3))
00:23:26 - should also work when the first bus is late *** FAILED *** (4 milliseconds)
00:23:26   akka.actor.InvalidActorNameException: actor name [BeamMobsim.iteration] is not unique!
00:23:26   at akka.actor.dungeon.ChildrenContainer$NormalChildrenContainer.reserve(ChildrenContainer.scala:134)
00:23:26   at akka.actor.dungeon.Children.reserveChild(Children.scala:137)
00:23:26   at akka.actor.dungeon.Children.reserveChild$(Children.scala:135)
00:23:26   at akka.actor.ActorCell.reserveChild(ActorCell.scala:445)
00:23:26   at akka.testkit.TestActorRef.<init>(TestActorRef.scala:28)
00:23:26   at akka.testkit.TestActorRef$.apply(TestActorRef.scala:161)
00:23:26   at akka.testkit.TestActorRef$.apply(TestActorRef.scala:151)
00:23:26   at beam.agentsim.agents.PersonAgentSpec.$anonfun$new$15(PersonAgentSpec.scala:535)
00:23:26   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
00:23:26   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
```
Issue at akka github: https://github.com/akka/akka/issues/24165#issuecomment-360046650. I understand it like we shouldn't have the same actor name across different test, but in our test we violate this rule.

@JustinPihony  what do you think?

### PR DESCRIPTION
**Pull request by dimaopen**
Moved stopping of [BeamMobsim.iteration] actor to afterEach method. This allows stopping this actor in case the test method fails. Not all tests may pass, but these failures are not related to this change.

Fixes #2514 

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2588)

### CODE REVIEW

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by REASY** [comment]
test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

File: src/test/scala/beam/agentsim/agents/PersonAgentSpec.scala [hunk 1]
```diff
@@ -860,4 +857,14 @@ class PersonAgentSpec
     super.afterAll()
   }
 
+  override def afterEach(): Unit = {
+    import scala.language.postfixOps
+    if (iteration != null) {
```
Comment (REASY) [inline src/test/scala/beam/agentsim/agents/PersonAgentSpec.scala:862]: nitpick: we're in Scala, maybe better to handle it via `Option` type?

Comment (dimaopen) [reply src/test/scala/beam/agentsim/agents/PersonAgentSpec.scala]: Done

**Comment by dimaopen** [comment]
Test!

**Comment by dimaopen** [comment]
Test!

**Pull request merged**